### ISSUE TITLE
Logic reversed in getAttribute()

### ISSUE THREAD

**Issue by valorin** [kirkbushell/eloquence#59]
I've just noticed an interesting discrepancy in `getAttribute()` between how the `CamelCasing` trait works and the original Eloquent Model.

Consider the [Eloquent Model](https://github.com/laravel/framework/blob/5.2/src/Illuminate/Database/Eloquent/Model.php#L2604-L2617) function:

```
public function getAttribute($key)
{
    if (array_key_exists($key, $this->attributes) || $this->hasGetMutator($key)) {
        return $this->getAttributeValue($key);
    }
    return $this->getRelationValue($key);
}
```

Now look at [CamelCasing](https://github.com/kirkbushell/eloquence/blob/master/src/Behaviours/CamelCasing.php#L27-L40):

```
public function getAttribute($key)
{
    if (method_exists($this, $key)) {
        return $this->getRelationValue($key);
    }
    return parent::getAttribute($this->getSnakeKey($key));
}
```

Eloquent allows attributes (even read-only ones) precedence over relations. CamelCasing favours relationships first.

It seems to me that following Eloquence's order of preference would be a good idea, so the behaviour is consistent between both.

**Comment by kirkbushell** [comment]
@valorin cheers! i have a feeling this may have been changed in the core codebase, as I'm fairly certain I duplicated the first bit of it. Something in the back of my mind is saying that I felt it was weird to begin with (placing relations over attributes). 

Happy to change this - it should definitely maintain the same behaviour as Eloquent.

**Comment by bicatu** [comment]
any chance this will be updated?

**Comment by kirkbushell** [comment]
Certainly, soon as I have time.

**Comment by askmrsinh** [comment]
Anyone who is surprised by this, please have a look at the new function code (10.x): https://github.com/laravel/framework/blob/b4ab62986000eaa411a22dbeb112b21c10420fdc/src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php#L428

**Comment by kirkbushell** [comment]
This is resolved in the coming 11.0 release.
